### PR TITLE
Complete user names too

### PR DESCRIPTION
**Pull request by purcell**
This commit adds a second completion-at-point function which completes `@user` references.  This initial version simply grabs all the users who opened issues or pull requests, since this information will be quickly available locally. Finding a better source of user info would be a nice enhancement, but this is already useful in my own projects.

### CODE REVIEW

File: RelNotes/0.1.6.org [hunk 1]
```diff
@@ -87,10 +87,11 @@
...
   for the current branch.  [[PR:269]]
-- Completion of issue numbers, e.g. "#123" is supported in edit and
-  commit message buffers via the standard ~completion-at-point~
-  mechanism, and therefore also via ~company~'s ~capf~ backend.  Call
-  ~magithub-completion-enable-everywhere~ to enable.  [[PR:263]]
+- Completion of issue numbers ("#123"), and user names ("@purcell") is
+  supported in edit and commit message buffers via the standard
+  ~completion-at-point~ mechanism, and therefore also via ~company~'s ~capf~
+  backend.  This is enabled by default in certain buffers via the
+  ~magithub-features~ mechanism.  [[PR:263]]
```
Comment (vermiculus) [inline RelNotes/0.1.6.org]: We could probably mention this PR in the release notes, too 👍

File: magithub-completion.el [hunk 2]
```diff
@@ -62,12 +62,43 @@ Add this to `completion-at-point-functions' in buffers where you want this to be
...
+            completions)
+        (dolist (i (magithub--issue-list))
+          (let-alist i
+            (when (string-prefix-p prefix .user.login)
+              (let ((candidate (copy-sequence .user.login))
+                    (association (and .author_association
+                                      (not (string= "NONE" .author_association))
+                                      .author_association)))
+                (set-text-properties 0 (length candidate) (list :user .user) candidate)
+                (set-text-properties 0 (length candidate) (list :association association) candidate)
```
Comment (vermiculus) [inline magithub-completion.el]: won't this line completely remove the `:user` property set above?

Comment (vermiculus) [reply magithub-completion.el]: We could probably just `propertize` it as it's pushed to `completions`.

File: magithub-completion.el [hunk 3]
```diff
@@ -62,12 +62,43 @@ Add this to `completion-at-point-functions' in buffers where you want this to be
...
+                                      (magithub-user-visit (get-text-property 0 :user c))))
+              )))))
 
 ;;;###autoload
 (defun magithub-completion-enable ()
   "Enable completion of info from magithub in the current buffer."
-  (add-to-list (make-local-variable 'completion-at-point-functions)
-               'magithub-completion-complete-issues))
+  (dolist (f '(magithub-completion-complete-issues magithub-completion-complete-users))
+    (add-to-list (make-local-variable 'completion-at-point-functions) f)))
```
Comment (vermiculus) [inline magithub-completion.el]: Do we need to `(make-local-variable 'completion-at-point-functions)` each time through the loop?

File: magithub-completion.el [hunk 4]
```diff
@@ -62,12 +62,43 @@ Add this to `completion-at-point-functions' in buffers where you want this to be
                                       (magithub-issue-visit (get-text-property 0 :issue c))))
               )))))
 
+;;;###autoload
+(defun magithub-completion-complete-users ()
+  "A `completion-at-point' function which completes \"@user\" user references.
+Add this to `completion-at-point-functions' in buffers where you
+want this to be available.  The user list is currently simply the
+list of all users who created issues or pull requests."
```
Comment (vermiculus) [inline magithub-completion.el]: Keep in mind this will only get issues/PRs that are currently open. I'm working on a way to make this more…better :-)

Comment (purcell) [reply magithub-completion.el]: Yeah, definitely a limitation that it would be nice to address.

File: magithub-completion.el (hunk 2 above)
Comment (purcell) [reply magithub-completion.el]: Yes and yes. I'll change that.

File: magithub-completion.el (hunk 3 above)
Comment (purcell) [reply magithub-completion.el]: Nope, true. Will fix.

**Comment by purcell** [comment]
Done.

**Pull request merged**